### PR TITLE
Move EntitySwitch component into plugin-catalog-react

### PR DESCRIPTION
**Pull request by alex-mckay**
## Hey, I just made a Pull Request!

Per @iamEAP's request, moving the EntitySwitch component from plugin-catalog to plugin-catalog-react. This will allow other plugins that need to use the component to not depend on catalog itself. 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/bright-kings-relate.md
Comment (alex-mckay) [inline .changeset/bright-kings-relate.md:1]: Happy to update this if the format is off! The [reviewing guide ](https://github.com/backstage/backstage/blob/master/REVIEWING.md)cites combining changesets for _unrelated_ information as bad form, but I thought as these changes are intertwined it made sense to connect them for easier traceability around when the move happens.

Omitted the example-app changeset since it's private, and omitted core-compat-api & create-app as my interpretation was they were internal changes, but less confident about those two/can imagine an argument that they should be changesets as well.

Comment (awanlin) [reply .changeset/bright-kings-relate.md:1]: You'll for sure need a changeset for `create-app` 👍

**Review by awanlin — CHANGES_REQUESTED** [review]
Hi @alex-mckay, what you're going to want to do here is deprecate the current version first. You can still add it to `plugin-catalog-react`. This allows Adopters to migrate on their own time versus having a breaking change. If we went forward with this it would have a pretty large impact considering everyone's Backstage instance uses this.

**Comment by alex-mckay** [comment]
Thanks @awanlin! I was trying to avoid letting the logic get out of sync between the two/figured the update was small enough to be low lift - but definitely hear you on giving consumers more time to make the change. Still slightly unsure about the changeset format (Is it OK to group all three like that? If not, do you separate them into different markdown files or just multiple '---' delineated sections in the same one?) - but let me know if this looks better!

File: .changeset/bright-kings-relate.md
Comment (iamEAP) [inline .changeset/bright-kings-relate.md:4]: When the deprecated code is eventually removed, it'll definitely be `major`, but perhaps `patch` (instead of `minor`) on `plugin-catalog` is more appropriate for now, since the API surface area isn't really changing.

File: .changeset/bright-kings-relate.md
Comment (iamEAP) [reply .changeset/bright-kings-relate.md:1]: > _Omitted the example-app changeset since it's private, and omitted core-compat-api_

Yep!  That makes sense to me.

> _as these changes are intertwined it made sense to connect them for easier traceability around when the move happens._

I think it definitely makes sense for `create-app` and `plugin-catalog` to include these details.  If there's any opportunity to reduce redundant info, it might be on `plugin-catalog-react`, where you could maybe get away with a one-liner like...

```md
---
'@backstage/plugin-catalog-react': minor
---

This package is now the source of truth for the `<EntitySwitch>` component, as well as the related utilities `isComponentType()`, `isEntityWith()`, `isKind()`, `isNamespace()`, and `isResourcetype()`.
```

File: plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx
Comment (iamEAP) [inline plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx:65]: Another way to potentially handle this is to just re-export these components/functions from `@backstage/plugin-catalog-react` (but in a way where it's still clear via `api-reports.md` and typescript that the import is `deprecated`).

**Review by iamEAP — COMMENTED** [review]
Thanks for tackling this @alex-mckay!  I offer a couple of comments, but will leave it to @backstage/catalog-maintainers for their approval/say.

File: plugins/catalog-react/src/components/EntitySwitch/EntitySwitch.tsx
Comment (Rugvip) [inline plugins/catalog-react/src/components/EntitySwitch/EntitySwitch.tsx:1]: We'll want to avoid duplicating this code, and instead re-exporting from the plugin. See for example https://github.com/backstage/backstage/blob/8a42303a0e4eea955785a9860bde3f78b4d691bf/plugins/home/src/deprecated.ts#L29-L81

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 🎉 

Could you elaborate a bit more on why and how you want to use these in other plugins? The original intention for these utilities where for them only to be used in the wiring in the app on the entity page.

File: plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx
Comment (alex-mckay) [reply plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx]: Good shout/did this following Patrik's suggestion as well!

**Comment by alex-mckay** [comment]
@Rugvip we're building custom entity pages for the new set of data experience plugins!

**Comment by awanlin** [comment]
@alex-mckay, did you mean to close this?

**Comment by awanlin** [comment]
Ok, cool, thanks for following up @alex-mckay 🚀

**Pull request closed without merging**
